cinder/scheduler/manager.py

```diff
@@ -471,15 +472,56 @@ def _extend_volume_set_error(self, context, ex, request_spec):
                 {'size': new_size - volume.size})
             volume_rpcapi.VolumeAPI().extend_volume(context, volume, new_size,
                                                     reservations)
-        except exception.NoValidBackend as ex:
-            QUOTAS.rollback(context, reservations,
-                            project_id=volume.project_id)
-            _extend_volume_set_error(self, context, ex, request_spec)
-            self.message_api.create(
-                context,
-                message_field.Action.EXTEND_VOLUME,
-                resource_uuid=volume.id,
-                exception=ex)
+        except exception.NoValidBackend:
+            try:
+                self._extend_migrate(context, volume, new_size, request_spec,
+                                     filter_properties, reservations)
+            except exception.NoValidBackend as ex:
+                QUOTAS.rollback(context, reservations,
+                                project_id=volume.project_id)
+                _extend_volume_set_error(self, context, ex, request_spec)
+                self.message_api.create(
+                    context,
+                    message_field.Action.EXTEND_VOLUME,
+                    resource_uuid=volume.id,
+                    exception=ex)
+
+    def _extend_migrate(self, context, volume, new_size, request_spec,
+                        filter_properties, reservations):
+
+        if volume.consistencygroup_id or volume.group_id:
+            raise exception.NoValidBackend(
+                reason='The volume is in a group and cannot be migrated.')
+
+        scheduler_hints = \
+            volume_utils.get_scheduler_hints_from_volume(volume)
+        filter_properties.update(scheduler_hints)
+        filter_properties.pop('new_size')
+
+        if not request_spec:
+            request_spec = {'volume_properties': {'size': new_size}}
+        else:
+            request_spec['volume_properties']['size'] = new_size
```

Comment:
Isn't this the code we had to remove from my patch to get the scheduler to see a difference between the new size and original size?

Reply:
This property is picked up by capacity filter, which needs to consider the new size when selecting a host to fit it.